localstack/utils/bootstrap.py

```diff
@@ -48,7 +48,7 @@
     "es": ["opensearch"],
     "cloudformation": ["s3", "sts"],
     "lambda": ["s3", "sqs", "sts"],
-    "firehose": ["kinesis"],
+    "firehose": ["kinesis", "opensearch", "es"],
```

Comment:
I can try to articulate better my point around mixing those dependencies up, with examples. 

Right now, this variable means: this service is _built_ on top on those. 
And this PR will change the meaning of this: it will then mean this service is either _built_ on top or _interact_ with this service, what we could also refer as "required" vs "optional", but I maybe see it in a different way that those words.

For the test selection, this would not work for 2 of my services: S3 and SNS. 

We would like to keep the strict loading for both of them, as they are compatible with it.

S3 has S3 notifications integration and a KMS integration: this would interact with the following services: SQS, Lambda (who needs `sts`), SNS, EventBridge, KMS. SNS would then need SES and Firehose, which then needs Opensearch and Kinesis. 
And the S3 image uses the strict service loading, but now it has 10 (or 11 with `es`) services more. 

Same for SNS, it is compatible with strict service loading, but when setting it up for the tests, it would now need: SQS, SES, Lambda, Firehose etc... same as above.

So merging those 2 variables in one has pretty huge repercussion in my opinion. 
Either if we need to add a level to the dependencies, "required" or "optional", or have 2 different variables, this I don't know. 

\cc @thrau @steffyP

Reply:
I agree with @bentsku. We are mixing up two concepts when we just "re-use" this `API_DEPENDENCIES`. 

For the testing approach there should be a separate list IMO.

Reply:
Definitely, I'm happy to implement this in here directly.
@thrau suggested `API_DEPENDENCIES_OPTIONAL` just next to `API_DEPENDENCIES`.
We could just use a merged version of the two in the test selection for now.
It should also be a bit restructured in upcoming iterations (@dominikschubert already mentioned that in a TODO), but I think that could be a good next step.
Would that suffice for now, @bentsku @steffyP?

Reply:
Once the change is there, I'll add SNS in the opt-in! S3 is everywhere, so I don't mind doing a full run 😄

Reply:
I added `API_DEPENDENCIES_OPTIONAL` in the last commit. Let me know what you think 😄

Reply:
Nice, that's neat 👍 I like the `defaultdict` with `set` 😉 awesome, thank you so much! I'll push SNS soon then 😄